api/envoy/api/v2/core/protocol.proto

```diff
@@ -216,6 +216,9 @@ message Http2ProtocolOptions {
   //
   // See `RFC7540, sec. 8.1 <https://tools.ietf.org/html/rfc7540#section-8.1>`_ for details.
   bool stream_error_on_invalid_http_messaging = 12;
+
+  // Specify duration to keep alive HTTP/2 stream. If not specified, this value is not set.
```

Comment:
Can you add to docs such as https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/timeouts.html details on how this interacts with existing timeouts?

I think I grok the idea here, that we're expressing an HTTP/2 level stream timeout, vs. a more abstract session level request timeout, but the two are closely related and it would help users to know what the motivation is and when to use one vs. another.

Reply:
I don't think this is HTTP/2 specific. It could be used on HTTP/1 long polls also. Can we move this to generic HTTP configuration and document as @htuch recommends? This also needs release notes, etc.